include/seqan3/contrib/parallel/buffer_queue.hpp

```diff
@@ -410,7 +390,7 @@ inline queue_op_status buffer_queue<value_t, buffer_t, buffer_policy>::try_pop(v
 
         newHeadReadPos = cyclic_increment(headReadPos, cap, roundSize);
 
-        if (this->headReadPos.compare_exchange_weak(headReadPos, newHeadReadPos))
+        if (this->headReadPos.compare_exchange_weak(headReadPos, newHeadReadPos, std::memory_order_acq_rel))
```

Comment:
Why are you changing this from the default `std::memory_order_seq_cst`?

Reply:
Because, it should be safe to acquire and release on the atomic, while it is more efficient on x86 architectures. `memory_order_seq_cst` will force a memory fence and thus can hit performance.

Reply:
I don't have any expertise on memory order. Do you have a nice little tutorial that describes the difference?

> it should be safe to acquire and release on the atomic

I'm not sure `memory_order_acq_rel` has anything to do with being "safe" to acquire or release an atomic. I think it has something to do with the ability of the compiler to reorder the code positions of the assembly instructions.

Reply:
Well, I haven't had a very good tutorial. Some blogs etc. but in general you are right. I should at first make it as restricive as possible and guarantee sequential ordering. One needs to invest more time to completely reason about the correct behaviour. I change it back